packages/framework-types/src/concepts/command.ts

```diff
@@ -2,16 +2,24 @@ import { Register } from './register'
 import { Class, PropertyMetadata } from '../typelevel'
 import { RoleAccess } from './role'
 
-/**
- * A `Command` is an imperative intent in your system.
- * All Command classes of your application must extend this class.
- */
-export interface CommandInterface {
-  handle(register: Register): Promise<void>
+export type CommandInterface = unknown
+
+export interface CommandClassInterface<TCommand = unknown> extends Class<TCommand> {
```

Comment:
I think you don't need this extra `CommandClassInterface` and you can leave the `CommandInterface` class as it was before (with the change of `handle` receiving the command as first argument). 
The key here is to apply the restriction of `CommandInterface` to the class, instead of the instance. See this comment LINK to see how it is applied.

Reply:
I don't see any links @alvaroloes 🤔

Reply:
Updated!

Reply:
Fixed in https://github.com/boostercloud/booster/pull/333/commits/47bb8bfdefe0210d9e7efe72065084d714e5ab58